demosplan/DemosPlanCoreBundle/ResourceTypes/BrandingResourceType.php

```diff
@@ -78,4 +81,47 @@ protected function getAccessConditions(): array
     {
         return [];
     }
+
+    /**
+     * @param Branding $updateTarget
+     */
+    public function getUpdatableProperties(object $updateTarget): array
+    {
+        if (!$this->currentUser->hasPermission('area_customer_settings')) {
+            return [];
+        }
+
+        $currentCustomerBrandingId = $this->currentCustomerService->getCurrentCustomer()->getBranding()?->getId();
+        if (null === $currentCustomerBrandingId) {
+            return [];
+        }
+
+        if ($currentCustomerBrandingId !== $updateTarget->getId()) {
+            return [];
+        }
+
+        $properties = [
+            $this->logo,
+        ];
+
+        if ($this->currentUser->hasPermission('feature_customer_branding_edit')) {
+            $properties[] = $this->styling;
+        }
+
+        return $this->toProperties(...$properties);
+    }
+
+    /**
+     * @param Branding $object
+     */
+    public function updateObject(object $object, array $properties): ResourceChange
+    {
+        $updater = new PropertiesUpdater($properties);
+        $updater->ifPresent($this->logo, $object->setLogo(...));
+        $updater->ifPresent($this->styling, $object->setCssvars(...));
```

Comment:
a property constraint was added in the Branding entity for CssVars 'ValidCssVarsConstraint'. Do u want to use it to validate the CssVars here ?

Reply:
This gets validated in row 123 `        $this->resourceTypeService->validateObject($object);`

Reply:
it's not a concern anymore the entity will be validated